pkg/epp/flowcontrol/registry/shard.go

```diff
@@ -30,74 +30,72 @@ import (
 //
 // # Role: The Data Plane Slice
 //
-// It represents a single, concurrent-safe slice of the registry's total state. It provides a read-optimized view for a
-// `controller.FlowController` worker.
+// It represents a single, concurrent-safe slice of the registry's total state, acting as an independent, parallel
+// execution unit. It provides a read-optimized view for a `controller.FlowController` worker, partitioning the overall
+// system state to enable horizontal scalability.
 //
-// # Concurrency: `RWMutex` and Atomics
+// # Concurrency Model: `RWMutex` for Topology, Atomics for Stats
 //
-// The `registryShard` balances read performance with write safety:
+// The `registryShard` balances read performance with write safety using a hybrid model:
 //
-//   - `sync.RWMutex` (mu): Protects the shard's internal maps (`priorityBands`) during administrative operations.
-//
-//   - Atomics (Stats): Aggregated statistics (`totalByteSize`, `totalLen`) use atomics for lock-free updates during
-//     delta propagation.
-//
-//   - Atomic Lifecycle (Status): The lifecycle state is managed via an atomic `status` enum.
+//   - `mu (sync.RWMutex)`: Protects the shard's internal topology (the maps of priority bands and queues) during
+//     administrative operations like flow registration, garbage collection, and configuration updates.
+//     Read locks are used on the hot path to look up queues, while write locks are used for infrequent structural
+//     changes.
+//   - Atomics: Aggregated statistics (`totalByteSize`, `totalLen`, etc.) and the `isDraining` flag use atomic
+//     operations, allowing for high-frequency, lock-free updates and reads of the shard's status and load, which is
+//     critical for the performance of the data path and statistics propagation.
 type registryShard struct {
+	// --- Immutable Identity & Dependencies (set at construction) ---
 	id     string
 	logger logr.Logger
 
-	// config holds the partitioned configuration for this shard, derived from the `FlowRegistry`'s global `Config`.
-	// It contains only the settings and capacity limits relevant to this specific shard.
-	config *ShardConfig
-
-	// status tracks the lifecycle state of the shard (Active, Draining, Drained).
-	// It is stored as an `int32` for atomic operations.
-	status atomic.Int32 // `componentStatus`
+	// onStatsDelta is the callback used to propagate statistics changes up to the parent registry.
+	onStatsDelta propagateStatsDeltaFunc
+	// orderedPriorityLevels is a cached, sorted list of priority levels.
+	orderedPriorityLevels []uint
 
-	// parentCallbacks provides the communication channels back to the parent registry.
-	parentCallbacks shardCallbacks
+	// --- State Protected by `mu` ---
 
-	// mu protects the shard's internal maps (`priorityBands`).
+	// mu protects the shard's internal topology (`priorityBands`) and `config`.
+	// TODO: This is a priority inversion issue. Administrative operations (e.g., GC) for a low-priority flow block all
```

Comment:
Will tackle this in a fast-followup PR once the flow control system is wired up and benchmarked.